File: reduks/src/main/java/com/beyondeye/reduks/StoreEnhancer.java
package com.beyondeye.reduks;

/**
 * get a store creator and return a new enhanced one
 * see https://github.com/reactjs/redux/blob/master/docs/Glossary.md#store-enhancer
 *
 * keep this interface in java to leverage kotlin SAM conversion (lambda automatical converted to implementation of java interface)
 * Created by daely on 8/23/2016.
 */
public interface StoreEnhancer<S> {
    StoreCreator<S> enhance(StoreCreator<S> next);
}

File: reduks/src/main/kotlin/com/beyondeye/reduks/StoreEnhancer.kt
package com.beyondeye.reduks

/**
 * get a store creator and return a new enhanced one
 * see https://github.com/reactjs/redux/blob/master/docs/Glossary.md#store-enhancer

 * single method interface, mainly used because kotlin does not support yet type alias for function types
 * Created by daely on 8/23/2016.
 */
interface StoreEnhancer<S> {
    fun enhance(next: StoreCreator<S>): StoreCreator<S>
}
